src/components/NavBar/index.js

```diff
@@ -48,238 +83,107 @@ const Container = styled.div`
-  border-radius: ${constants.borderRadius.large};
-  overflow: hidden;
+  padding-top: 2px;
+  padding-bottom: 2px;
 `;
 
-const LinkContainer = styled.span.attrs({ className: "linkContainer" })`
-  ${largeAndUp`
-    position: relative;
-    &:hover .linkList {
-      display: block;
-      position: absolute;
-      left: 0;
-      top: 41px;
-
-      .linkItem {
-        display: block;
-        padding-top: 8px;
-        padding-bottom: 8px;
-      }
-    }
-  `};
-`;
-
-const StyledLink = styled.a.attrs({ className: "linkItem" })`
-  display: inline-block;
-  text-decoration: none;
-  padding: 19px ${spacing.gutters.small}px 19px ${spacing.gutters.small}px;
-  ${mediumAndUp`
-    &:hover {
-      background-color: rgba(38, 38, 38, 0.1);
-    }
-    .inverted &:hover {
-      background-color: ${colors.shale};
-    }
-  `};
-`;
-
-const Link = ({ children, ...props }) => {
-  const [label, ...other] = Children.toArray(children);
-  return (
-    <LinkContainer>
-      <StyledLink {...props}>{label}</StyledLink>
-      {other}
-    </LinkContainer>
-  );
-};
-
-Link.propTypes = {
-  children: PropTypes.node
-};
-
-Link.defaultProps = {
-  children: null
-};
-
-const NavBar = ({ children, fixed, invert, ...props }) => (
-  <Nav {...props} fixed={fixed} invert={invert}>
+const NavBar = ({
+  children,
+  position,
+  invert,
+  className,
+  backgroundColor,
+  style,
+  message,
+  ...props
+}) => (
+  <Nav
+    {...props}
+    className={classNames(
+      position && `nav--${position}`,
+      invert && "nav--inverted",
+      !!backgroundColor && "nav--overlay",
+      className
+    )}
+    invert={invert}
+    style={{ ...style, backgroundColor }}
+  >
+    {message ? (
+      <MessageContainer>
+        <Row>
+          <Column role="note">{message}</Column>
+        </Row>
+      </MessageContainer>
+    ) : null}
     <Container>{children}</Container>
   </Nav>
 );
 
 NavBar.propTypes = {
   children: PropTypes.node,
-  fixed: PropTypes.bool,
-  invert: PropTypes.bool
+  className: PropTypes.string,
+  position: PropTypes.oneOf(["relative", "absolute", "fixed"]),
+  invert: PropTypes.bool,
+  message: PropTypes.node,
+  backgroundColor: PropTypes.string,
+  style: PropTypes.objectOf(PropTypes.string)
 };
 
 NavBar.defaultProps = {
+  className: null,
   children: null,
-  fixed: false,
-  invert: false
+  position: "relative",
+  invert: false,
+  message: null,
+  backgroundColor: null,
+  style: {}
 };
 
-NavBar.MenuButton = MenuButton;
-NavBar.Button = Button;
-NavBar.LinkRow = LinkRow;
-NavBar.LinkList = LinkList;
-NavBar.Link = Link;
+NavBar.MenuButton = Buttons.MenuButton;
+NavBar.SearchButton = Buttons.SearchButton;
+NavBar.UserButton = Buttons.UserButton;
+NavBar.TextButton = Buttons.TextButton;
+NavBar.LinkRow = Links.LinkRow;
+NavBar.LinkList = Links.LinkList;
+NavBar.Link = Links.Link;
 NavBar.Right = Right;
-NavBar.LogoContainer = LogoContainer;
+NavBar.Left = Left;
+NavBar.LogoContainer = Buttons.LogoContainer;
```

Comment:
Could these Buttons, LinkRow, LinkList, Link, Right, Left, and LogoContainer be applicable to other contexts? Unsure if they should all be lumped into the NavBar long-term if there is potential for reuse elsewhere and a reduction in the NavBar component's default size.